django_react_proj/users-fe/src/components/ConfirmRemovalModal.js

```diff
@@ -1,53 +1,56 @@
-import React, { Component, Fragment } from "react";
+import React, { useState, Fragment } from "react";
 import { Modal, ModalHeader, Button, ModalFooter } from "reactstrap";
 import axios from "axios";
 import { API_URL } from "../constants";
 
-class ConfirmRemovalModal extends Component {
-  state = {
-    modal: false
-  };
+const ConfirmRemovalModal = (props) => {
+  const [modalFlag, setModalFlag] = useState(
+        false
+  )
 
-  toggle = () => {
-    this.setState(previous => ({
-      modal: !previous.modal
-    }));
+  const toggle = () => {
+    setModalFlag(previous => (
+       !previous
+     ));
   };
+  
+  const handlerDeleteUser = async id => {
+    try {
+      await props.deleteUser(id)
+      toggle() 
+    } catch (error) {
+      console.log(error)
+    } 
 
-  deleteUser = id => {
-    axios.delete(API_URL + id).then(() => {
-      this.props.resetState();
-      this.toggle();
-    });
-  };
+   };
+  //gppp
+
+  return (
+    <Fragment>
+      <Button color="danger" onClick={() => toggle()}>
+        Usuń
+      </Button>
+      <Modal isOpen={modalFlag} toggle={toggle}>
+        <ModalHeader toggle={toggle}>
+          Czy chcesz usunąć konto?
+        </ModalHeader>
 
-  render() {
-    return (
-      <Fragment>
-        <Button color="danger" onClick={() => this.toggle()}>
-          Usuń
-        </Button>
-        <Modal isOpen={this.state.modal} toggle={this.toggle}>
-          <ModalHeader toggle={this.toggle}>
-            Czy chcesz usunąć konto?
-          </ModalHeader>
+        <ModalFooter>
+          <Button type="button" onClick={() => toggle()}>
+            Anuluj
+          </Button>
+          <Button
+            type="button"
+            color="primary"
+            onClick={() => handlerDeleteUser(props.id)}
```

Comment:
You can just pass func ref here since `id` is in global props (`onClick={handlerDeleteUser}`). Then you also need to change the function (take id from props istead of arguments).